src/freenet/client/async/SplitFileInserterStorage.java

```diff
@@ -218,7 +223,16 @@
 
     private final int overallStatusLength;
     
+    /** Protects cooldown calculations and noBlocksToSend. You MUST NOT take this lock while 
+     * holding any other locks!!!
+     * Cooldown calculation is tricky! Two major issues:
+     * 1. Consistency is essential because if we return Long.MAX_VALUE from getWakeupTime(), the
+     * insert will stall until it is explicitly rescheduled. That's the whole point, in fact.
```

Comment:
How does deferring the request until explicitly rescheduled interact with consistency? I don't think I understand.